# Question 56

This implementation of pow uses exceptions to communicate between an auxiliary method and a main method when an answer has been reached. It multiplies until an answer has been reached, at which point the auxiliary method throws an exception which the main method catches and returns the answer. This is effective at communicating when a process has terminated between different methods, but isn't really what exceptions were designed to do and is therefore bad form and is somewhat unreadable. A better solution would be recursion.